### PR TITLE
Implemented some JS elements to website

### PR DESCRIPTION
**Pull request by caroljli**
**Features:**
* Scroll up toggled when scrolling.
* Picture gallery slider.
* Navbar to list (for ~ minimalism ~)
* Post content
* Fade in for landing page
* Back button

**Screenshots:** (not everything -- some things are animations)
Screenshot 2020-05-28 at 5 47 34 PM
* Nav bar that opens and closes when clicking the + button.
Screenshot 2020-05-28 at 5 47 56 PM
* Image slider for food pictures, appearance of back to top while scrolling.
Screenshot 2020-05-28 at 5 48 10 PM
* Absence of back to top when at top of page.
image
* The modal when clicked on -- contains image gallery.

**Notes:**
* I'm sorry that this PR is so large. I forgot to checkout to a new branch when I made changes today (ie. the modal) because I'm used to seeing the branch I'm working on (via iTerm haha) but the Cloud Shell terminal doesn't indicate the branch and I had forgotten to check out earlier.
* I'm including screenshots of the modal functionality, but will probably move on to a different PR for other things. I'm sorry about the lack of organization.

### CODE REVIEW

File: portfolio/src/main/webapp/assets/script.js [hunk 1]
```diff
@@ -130,17 +101,43 @@ function scrollUpHelper() {
     var top = document.getElementById("top-button");
     if (document.documentElement.scrollTop > 20) {
         top.style.display = "block";
-    } else {
+    } else if (top) {
```
Comment (atupem) [inline portfolio/src/main/webapp/assets/script.js:104]: Why check for existence here?

File: portfolio/src/main/webapp/about.html [hunk 2]
```diff
@@ -119,4 +133,64 @@ <h3>for Passerine Cowbirds &nbsp; • &nbsp;<a class="light_blue">Pandas, Sci-Ki
...
+                                <img src="/assets/images/img-friends.jpg"></img>
+                                <div class="slideshow-caption">
+                                    my friends and i RIGHT BEFORE everything went virtual :'(
+                                </div>
+                            </div>
+
+                            <div class="slides trans">
+                                <img src="/assets/images/img-lin.jpg"></img>
+                                <div class="slideshow-caption">
+                                    my lin! fun fact: big and gbig are both rly cool dancers
```
Comment (aditij113) [inline portfolio/src/main/webapp/about.html:174]: lin --> line

(I think :))

File: portfolio/src/main/webapp/assets/script.js [hunk 3]
```diff
@@ -43,3 +18,126 @@ function collapseMenu() {
...
+function current(n) {
+    show(index = n);
+}
+
+function show(n) {
+    var i;
+    var slides = document.getElementsByClassName("slides");
+    var indices = document.getElementsByClassName("index");
+
+    if (n > slides.length) {
```
Comment (aditij113) [inline portfolio/src/main/webapp/assets/script.js:55]: Can you add a comment for what this and the next if statement are doing?

File: portfolio/src/main/webapp/assets/script.js [hunk 4]
```diff
@@ -43,3 +18,126 @@ function collapseMenu() {
...
+
+function move(n) {
+    show(index += n);
+}
+
+function current(n) {
+    show(index = n);
+}
+
+function show(n) {
```
Comment (aditij113) [inline portfolio/src/main/webapp/assets/script.js:50]: This function is doing enough things where I feel it deserves a function description.

File: portfolio/src/main/webapp/index.html [hunk 5]
```diff
@@ -8,25 +8,29 @@
...
-                </p>
-                </div>
-            <div class="column is-one-fifth"></div>
-            </div>
+        <div class="landing_text">
+        <h1>Hi, I'm Carol!</h1>
+        <p>
+            I'm Carol Li, a Canadian (🇨🇦) that likes to draw, design, and 
+            develop. I'm currently an undergraduate student at the 
+            University of Pennsyl[vania.
```
Comment (aditij113) [inline portfolio/src/main/webapp/index.html:32]: fix spelling on Pennsylvania

File: portfolio/src/main/webapp/posts.html [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+                                enough when a grade didn't turn out how I wanted, or I couldn't solve a problem in a problem set,
+                                or I compared myself to everyone else and felt like I wasn't doing enough. As I did this, I looked
+                                over the things that I was able to accomplish, the new friends I made and the new opportunities and
+                                experiences I was able to have. As a consequence, I didn't know it at the time, but I was doing
+                                enough. 
+
+                                <br></br>
+
+                                On another note, I think something very characteristic of college is that time seems to be 
+                                condensed. Little of it is mundane,
```
Comment (aditij113) [inline portfolio/src/main/webapp/posts.html:177]: abrupt new line?

File: portfolio/src/main/webapp/posts.html [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+
+                                On another note, I think something very characteristic of college is that time seems to be 
+                                condensed. Little of it is mundane, 
+                                as there's always something to see or someone to meet, and you feel like you have to pack 
+                                a lot of experiences together in order to meet the pressure of being "sceney" or active in 
+                                your social life. Although I gave into that pressure a lot during freshmen year (a phenomenon 
+                                almost directed linked to PennFace, which is basically what Penn calls duck syndrome), I think 
+                                that, regardless of feeling like you have to be social, you feel bad when you quickly lose touch 
+                                of friends that you made just as quickly. I think its hard to dissociate sometimes and just 
+                                appreciate those around you.
```
Comment (aditij113) [inline portfolio/src/main/webapp/posts.html:184]: Thanks for sharing this :) 💯

File: walkthroughs/week-2-web-development/examples/stanley/src/main/webapp/script.js [hunk 8]
```diff
@@ -30,3 +30,5 @@ function randomizeImage() {
   imageContainer.innerHTML = '';
   imageContainer.appendChild(imgElement);
 }
+
+
```
Comment (aditij113) [inline walkthroughs/week-2-web-development/examples/stanley/src/main/webapp/script.js:34]: remove these blank lines if they're not needed

File: portfolio/src/main/webapp/assets/script.js (hunk 1 above)
Comment (caroljli) [reply portfolio/src/main/webapp/assets/script.js:104]: When I was checking the console.log output, it gave an element not found error without checking for existence -- I wasn't sure if that was an issue so I checked for existence.

File: portfolio/src/main/webapp/about.html (hunk 2 above)
Comment (caroljli) [reply portfolio/src/main/webapp/about.html:174]: oop they are my big/grandbig so its my my lineage :))

File: portfolio/src/main/webapp/index.html (hunk 5 above)
Comment (caroljli) [reply portfolio/src/main/webapp/index.html:32]: Fixed it in modal PR!

File: portfolio/src/main/webapp/posts.html (hunk 6 above)
Comment (caroljli) [reply portfolio/src/main/webapp/posts.html:177]: Fixed the spacing!

**Pull request closed without merging**